File: agent1/Backend/app/Services/PayloadProcessors/NetworkProcessor.php
<?php

declare(strict_types=1);

namespace App\Services\PayloadProcessors;

use App\Models\Machine;
use App\Models\MachineNetworkAdapter;
use Illuminate\Support\Facades\Log;

class NetworkProcessor
{
    public function process(Machine $machine, array $payload, $healthLog = null): void
    {
        try {
            $adapters = $payload['networkAdapters'] ?? $payload['network'] ?? $payload['network_adapters'] ?? [];
            if (empty($adapters)) {
                return;
            }

            foreach ($adapters as $adapter) {
                $adapterName  = $adapter['adapterName'] ?? $adapter['adapter_name'] ?? 'Unknown';
                $isConnected  = $adapter['isConnected'] ?? $adapter['is_connected'] ?? false;
                $ipV4         = $adapter['ipAddressV4'] ?? $adapter['ip_address_v4'] ?? $adapter['ipAddress'] ?? null;
                $mac          = $adapter['macAddress'] ?? $adapter['mac_address'] ?? null;
                $speed        = $adapter['connectionSpeedMbps'] ?? $adapter['connection_speed_mbps'] ?? null;
                $bytesSent    = $adapter['bytesSent'] ?? $adapter['bytes_sent'] ?? null;
                $bytesRecv    = $adapter['bytesReceived'] ?? $adapter['bytes_received'] ?? null;

                $ipV6 = $adapter['ipAddressV6'] ?? $adapter['ip_address_v6'] ?? null;
                $adapterType = $adapter['adapterType'] ?? $adapter['adapter_type'] ?? null;

                MachineNetworkAdapter::updateOrCreate(
                    [
                        'machine_id'   => $machine->id,
                        'adapter_name' => $adapterName,
                    ],
                    [
                        'ip_address'     => $ipV4,
                        'ip_address_v6'   => $ipV6,
                        'mac_address'    => $mac,
                        'adapter_type'   => $adapterType,
                        'speed'          => $speed,
                        'bytes_sent'     => $bytesSent,
                        'bytes_received' => $bytesRecv,
                        'status'         => $isConnected ? 'connected' : 'disconnected',
                        'updated_at'     => now(),
                    ]
                );
            }

            $totalSent = 0;
            $totalRecv = 0;
            foreach ($adapters as $adapter) {
                $totalSent += (int)($adapter['bytesSent'] ?? $adapter['bytes_sent'] ?? 0);
                $totalRecv += (int)($adapter['bytesReceived'] ?? $adapter['bytes_received'] ?? 0);
            }

            \App\Models\MachineCurrentStatus::updateOrCreate(
                ['machine_id' => $machine->id],
                [
                    'network_sent_bytes'     => $totalSent,
                    'network_received_bytes' => $totalRecv,
                    'collected_at'           => now(),
                ]
            );

            if ($healthLog) {
                $healthLog->update([
                    'network_sent_bytes'     => $totalSent,
                    'network_received_bytes' => $totalRecv,
                ]);
            }

            Log::debug('NetworkProcessor: Processed network data', [
                'machine_id' => $machine->id,
                'count'      => count($adapters),
            ]);
        } catch (\Throwable $e) {
            Log::error('NetworkProcessor::process - Failed', [
                'machine_id' => $machine->id,
                'error'      => $e->getMessage(),
                'trace'      => $e->getTraceAsString(),
            ]);
        }
    }
}

File: DeskGuardBackend/Services/PayloadProcessors/NetworkProcessor.cs
using System;
using System.Text.Json;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Logging;
using DeskGuardBackend.Data;
using DeskGuardBackend.Entities;
using DeskGuardBackend.Extensions;

namespace DeskGuardBackend.Services.PayloadProcessors
{
    public class NetworkProcessor : IPayloadProcessor
    {
        private readonly DeskGuardDbContext _dbContext;
        private readonly ILogger<NetworkProcessor> _logger;

        public NetworkProcessor(DeskGuardDbContext dbContext, ILogger<NetworkProcessor> logger)
        {
            _dbContext = dbContext;
            _logger = logger;
        }

        public async Task ProcessAsync(Machine machine, JsonElement payload, HealthLog healthLog)
        {
            try
            {
                var adaptersProp = payload.GetPropertyOrNull("networkAdapters") ?? payload.GetPropertyOrNull("network") ?? payload.GetPropertyOrNull("network_adapters");
                if (adaptersProp == null || adaptersProp.Value.ValueKind != JsonValueKind.Array) return;

                var adapters = adaptersProp.Value;

                long totalSent = 0;
                long totalRecv = 0;

                foreach (var adapter in adapters.EnumerateArray())
                {
                    var adapterName = adapter.GetStringProperty("adapterName") ?? adapter.GetStringProperty("adapter_name") ?? "Unknown";
                    var isConnected = adapter.GetBooleanProperty("isConnected") ?? adapter.GetBooleanProperty("is_connected") ?? false;
                    var ipV4 = adapter.GetStringProperty("ipAddressV4") ?? adapter.GetStringProperty("ip_address_v4") ?? adapter.GetStringProperty("ipAddress");
                    var mac = adapter.GetStringProperty("macAddress") ?? adapter.GetStringProperty("mac_address");
                    var speed = adapter.GetInt64Property("connectionSpeedMbps") ?? adapter.GetInt64Property("connection_speed_mbps");
                    var bytesSent = adapter.GetInt64Property("bytesSent") ?? adapter.GetInt64Property("bytes_sent") ?? 0;
                    var bytesRecv = adapter.GetInt64Property("bytesReceived") ?? adapter.GetInt64Property("bytes_received") ?? 0;
                    var adapterType = adapter.GetStringProperty("adapterType") ?? adapter.GetStringProperty("adapter_type");

                    totalSent += bytesSent;
                    totalRecv += bytesRecv;

                    var dbAdapter = await _dbContext.MachineNetworkAdapters
                        .FirstOrDefaultAsync(n => n.MachineId == machine.Id && n.AdapterName == adapterName);

                    if (dbAdapter == null)
                    {
                        dbAdapter = new MachineNetworkAdapter { MachineId = machine.Id, AdapterName = adapterName };
                        await _dbContext.MachineNetworkAdapters.AddAsync(dbAdapter);
                    }

                    dbAdapter.IpAddress = ipV4;
                    dbAdapter.MacAddress = mac;
                    dbAdapter.AdapterType = adapterType;
                    dbAdapter.Speed = speed;
                    dbAdapter.Status = isConnected ? "connected" : "disconnected";
                    dbAdapter.UpdatedAt = DateTime.UtcNow;
                }

                // Update MachineCurrentStatus
                var currentStatus = await _dbContext.MachineCurrentStatuses
                    .FirstOrDefaultAsync(s => s.MachineId == machine.Id);

                if (currentStatus == null)
                {
                    currentStatus = new MachineCurrentStatus { MachineId = machine.Id };
                    await _dbContext.MachineCurrentStatuses.AddAsync(currentStatus);
                }

                currentStatus.NetworkSentBytes = totalSent;
                currentStatus.NetworkReceivedBytes = totalRecv;
                currentStatus.CollectedAt = DateTime.UtcNow;

                // Update shared health log row
                healthLog.NetworkReceivedBytes = totalRecv;
                healthLog.NetworkSentBytes = totalSent;

                await _dbContext.SaveChangesAsync();

                _logger.LogDebug("NetworkProcessor: Processed network telemetry for machine {MachineId}", machine.Id);
            }
            catch (Exception ex)
            {
                _logger.LogError(ex, "NetworkProcessor: Failed to process network metrics for machine {MachineId}", machine.Id);
            }
        }
    }
}
